worker/draft.go

```diff
@@ -867,9 +869,18 @@ func (n *node) processApplyCh() {
 			ms := x.SinceMs(start)
 			ostats.RecordWithTags(context.Background(), tags, x.LatencyMs.M(ms))
 		}
-
 		n.Proposals.Done(prop.Key, perr)
 		n.Applied.Done(prop.Index)
+		if !ok && prop.Restore != nil && perr == nil {
```

Comment:
why do we check for `!ok`? Can we get duplicate restore proposals?

Reply:
Yes, we can get duplicate proposals (not specific to restore proposals). Here, https://github.com/dgraph-io/dgraph/blob/b7e7dad2791496800cc1e9fbe40e0d89214672a4/worker/proposal.go#L293-L297 we retry the proposal 3 times. And it might happen that we retried it even if it made to raft log on any server. So for safety, we use the same unique key for the identification of duplicates.